test/interface/sparsediff_tests.jl

```diff
@@ -34,22 +34,27 @@ jac_sp = sparse(generate_sparsity_pattern(10))
 colors = repeat(1:3,10)[1:10]
 u0=[1.,2.,3,4,5,5,4,3,2,1]
 tspan=(0.,10.)
-odefun_sp= ODEFunction(f,colorvec=colors,jac_prototype=jac_sp)
-prob_sp = ODEProblem(odefun_sp,u0,tspan)
-prob_std = ODEProblem(f,u0,tspan)
 
-sol_sp=solve(prob_sp,Rodas5(autodiff=false),abstol=1e-10,reltol=1e-10)
-@test sol_sp.retcode==:Success#test sparse finitediff
-sol=solve(prob_std,Rodas5(autodiff=false),abstol=1e-10,reltol=1e-10)
-@test sol_sp.u[end]≈sol.u[end] atol=1e-10
-@test length(sol_sp.t)==length(sol.t)
+odefun_std = ODEFunction(f)
+prob_std = ODEProblem(odefun_std,u0,tspan)
 
-sol_sp=solve(prob_sp,Rodas5(autodiff=false))
-sol=solve(prob_std,Rodas5(autodiff=false))
-@test sol_sp.u[end]≈sol.u[end]
-@test length(sol_sp.t)==length(sol.t)
-
-sol_sp=solve(prob_sp,Rodas5())
-sol=solve(prob_std,Rodas5())
-@test sol_sp.u[end]≈sol.u[end]
-@test length(sol_sp.t)==length(sol.t)
\ No newline at end of file
+for ad in [true, false]
+    for tol in [nothing, 1e-10]
+        sol_std=solve(prob_std,Rodas5(autodiff=ad),reltol=tol,abstol=tol)
+        @test sol_std.retcode==:Success
+        for prob in map(f->ODEProblem(f,u0,tspan),
+                        [ODEFunction(f,colorvec=colors,jac_prototype=jac_sp),
+                         ODEFunction(f,jac_prototype=jac_sp),
+                         #ODEFunction(f,colorvec=colors) # this one fails both the u[end] and length tests
```

Comment:
Any idea why the commented test would behave differently?

Reply:
Because one must have sparsity to get sparse jacobian work appropriately.

Reply:
Is this still the case now with our update?  Because some tests still behave erratically.